### PR TITLE
TTIEX355P Bug fixing

### PR DESCRIPTION
**Pull request by JamesKingWork**
### Description of work

TTIEX355P:

- Correct range controls to use DRVH and DRVL rather than HOPR and LOPR
- Autosave setpoints if macro set

### To test

https://github.com/ISISComputingGroup/IBEX/issues/6397

### Acceptance criteria

*List the acceptance criteria for the PR*

---

#### Code Review

- [ ] A copy of the manual has been placed on the shared drive
- [ ] Pertitent information has been stored in the [wiki](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Specific-Device-IOC)
- [ ] Does the IOC conform to IBEX standards?
    - [PV naming](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/PV-Naming).
    - [Disable records](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Disable-records)
    - [Record simulation](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Record-Simulation)
    - [Finishing touches](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/IOC-Finishing-Touches)
- [ ] If an OPI has been modified, does it conform to the [style guidelines](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/OPI-Creation)?
- [ ] Do the IOC and support module conform to the new [build guidelines](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Reducing-Build-Dependencies)
- [ ] Have the changes been recorded appropriately in a PR for [release notes](https://github.com/ISISComputingGroup/IBEX/blob/master/release_notes/ReleaseNotes_Upcoming.md)?
- [ ] Is the device's flow control setting correct? [For most devices flow control should be OFF](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Flow-control).

### Functional Tests

- IOC responds correctly in:
    - [ ] Devsim mode
    - [ ] Recsim mode
    - [ ] Real device, if available
- [ ] Supplementary IOCs (`..._0n` where `n>1`) run correctly
- [ ] Log files do not report undefined macros (serach for `macLib: macro` to find instances of `macLib: macro [macro name] is undefined...`

### Final steps

- [ ] Update the IOC submodule in the main EPICS repo. See [Git workflow](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Git-workflow) page for details.
- [ ] Reviewer has merged the associated PR for the [release notes](https://github.com/ISISComputingGroup/IBEX/blob/master/release_notes/ReleaseNotes_Upcoming.md)

### CODE REVIEW

File: TTIEX355P/iocBoot/iocTTIEX355P-IOC-01/st.cmd [hunk 1]
```diff
@@ -55,10 +55,12 @@ $(IFNOTFIELD_CONV_DEFINED) epicsEnvSet("AMPSTOGAUSS", "1")
 $(IFNOTFIELD_CONV_DEFINED) epicsEnvSet("FIELD_CONV_DEFINED", "0")
 $(IFFIELD_CONV_DEFINED) epicsEnvSet("FIELD_CONV_DEFINED", "1")
 
+stringiftest("DOAUTOSAVESETPOINTS", "$(AUTOSAVESETPOINTS=NO)", 5, "YES")
+
```
Comment (FreddieAkeroyd) [inline TTIEX355P/iocBoot/iocTTIEX355P-IOC-01/st.cmd:59]: Do you need "DO" ?  I feel  `IFAUTOSAVESETPOINTS`   sounds nicer than   `IFDOAUTOSAVESETPOINTS`

File: TTIEX355P/TTIEX355P-IOC-01App/Db/ttiEX355P.db [hunk 2]
```diff
@@ -42,15 +42,17 @@ record(ao, "$(P)VOLTAGE:SP")
     field(SIML, "$(P)SIM")
     field(SIOL, "$(P)SIM:VOLTAGE:SP")
     field(SDIS, "$(P)DISABLE")
+
+$(IFDOAUTOSAVESETPOINTS) info(autosaveFields, "VAL")
 }
```
Comment (FreddieAkeroyd) [inline TTIEX355P/TTIEX355P-IOC-01App/Db/ttiEX355P.db:47]: Would it be worth defaulting this e.g. $(IFAUTOSAVESETPOINTS=#)

**Review by DominicOram — CHANGES_REQUESTED** [review]
I agree with @FreddieAkeroyd's comments. Otherwise looks good!

**Review by DominicOram — APPROVED** [review]
Looks good, thanks

**Pull request merged**